integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/main/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverService.java

```diff
@@ -0,0 +1,367 @@
+                        Map<String, SparkContextState> tempSparkContextsStates = new HashMap<>();
+                        tempSparkContextsStates.putAll(sparkContextsStates);
+                        if (!sparkContextsStates.isEmpty()) {
+                            for (SparkContextState sparkContextState: tempSparkContextsStates.values()) {
+                                if (!sparkContextState.isLocked() && sparkContextState.hasTimedOut()) {
+                                    getLogger().info("Found context {} which timed out", new Object[]{sparkContextState.contextName});
+                                    deleteContext(sparkContextState.contextName);
+                                }
+                            }
+                        }
+                    }
+                } catch (Exception ex) {
+                    getLogger().trace(ex.getMessage(), ex);
+                }
+            }
+        };
+        long delay = 0;
+        long intervalPeriod = 10000;
+
+        // schedules the task to be run in an interval
+        contextTimeoutTimer.scheduleAtFixedRate(contextTimeoutTask, delay,
+                                                intervalPeriod);
+        getLogger().info("Created and scheduled context timeout task");
+    }
+
+    /**
+     * Stop timer task and kill all running Spark contexts.
+     */
+    @OnShutdown
+    public void shutdown() {
+        contextTimeoutTimer.cancel();
+        for (SparkContextState sparkContextState : sparkContextsStates.values()) {
+            deleteContext(sparkContextState.contextName);
+        }
+    }
+
+    /**
+     * Checks to see if a Spark Context is new on the Controller Service
+     *
+     * @param contextName the name of the Context
+     * @return true or false
+     */
+    private boolean isNewContext(String contextName) {
+        boolean newContext = false;
+        synchronized (sparkContextsStates) {
+            try {
+                boolean contextInitializing = true;
+                while (contextInitializing) {
+                    if (!sparkContextsStates.containsKey(contextName)) {
+                        getLogger().info("Context: {} is a new Context in Controller service", new Object[]{contextName});
+                        sparkContextsStates.put(contextName, new SparkContextState(contextName));
+                        newContext = true;
+                        contextInitializing = false;
+                    } else if (!sparkContextsStates.get(contextName).getRunningState()) {
+                        getLogger().info("Context: {} is being created... waiting", new Object[]{contextName});
+                        sparkContextsStates.wait();
+                    } else {
+                        contextInitializing = false;
+                    }
+                }
+            } catch (Exception ex) {
+                getLogger().trace(ex.getMessage(), ex);
+            }
+        };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)